File: python/panel.py
class Panel:
    def __init__(self,
                 index,
                 idle_color):
        self.interaction_active = False
        self.index = index
        self.idle_color = idle_color
        self.current_value = 0
        self.color = self.idle_color
        self.reset()

    def to_dict(self):
        return {
            'index': self.index,
            'interactionActive': self.interaction_active,
            'idle_color': self.idle_color.to_dict(),
            'currentValue': self.current_value,
            'color': self.color.to_dict()
        }

    def set_active(self, active):
        self.interaction_active = active

    def reset(self):
        self.current_value = 0
        self.color = self.idle_color

File: arduino/Aunisoma-Sketch/Panel.cpp
//
// Created by David Clay on 6/15/23.
//

#include "Panel.h"
#include "Color.h"

Panel::Panel(int index, Color idleColor): idleColor(0, 0, 0), color(0, 0, 0) {
    this->index = index;
    this->idleColor = idleColor;
    this->color = idleColor;
    this->active = false;
    this->currentValue = 0;
}

void Panel::reset() {
    this->color = this->idleColor;
    this->currentValue = 0;
}
